ios/Event+JSON.swift

```diff
@@ -130,3 +130,140 @@ extension TimedEventType {
 extension PlayEvent: TimedEventType {}
 extension PausedEvent: TimedEventType {}
 extension PlayingEvent: TimedEventType {}
+
+extension CueEvent where Self: Event {
+    func toJSON() -> [AnyHashable: Any] {
+        // Initial JS event object.
+        var json: [AnyHashable: Any] = [
+            "name": name,
+            "timestamp": timestamp,
+            "html": html,
+            "text": text,
+            "start": startTime,
+            "end": endTime
+        ]
+
+        // Initial JS cue object.
+        var cueJson: [AnyHashable: Any] = [
+            "text": text,
+            "html": html,
+            "start": startTime,
+            "end": endTime,
+        ]
+
+        // Compute cue image base64 if available.
+        if let data = image?.pngData() {
+            // Source string should be compatible with RN's `Image` base64 data format.
+            let imageSrc = "data:image/png;base64,\(data.base64EncodedString())"
+            json["image"] = imageSrc
+            cueJson["image"] = imageSrc
+        }
+
+        // Define CEA-only positioning properties.
+        if let position = position {
+            cueJson["ceaPosition"] = [
+                "row": position.row,
+                "column": position.column
+            ]
+        }
+
+        // Define VTT positioning properties.
+        if let cue = vtt {
+            // Box size
+            cueJson["size"] = cue.size
+
+            // Writing direction
+            cueJson["direction"] = {
```

Comment:
Actually, it would help me a lot to have the json representation of those objects out of the box. But I don't seem to find anything related to data encoding in the public docs, except for some shady undocumented functions like [`_toJsonData()`](https://cdn.bitmovin.com/player/ios/3/docs/Classes/SourceConfig.html#/c:@M@BitmovinPlayer@objc(cs)BMPSourceConfig(im)_toJsonData) which looks like internal stuff, so I didn't dare relying on them for JSON parsing.

Also, I renamed this `vertical` property to `direction` based on it says on [the docs](https://cdn.bitmovin.com/player/ios/3/docs/Classes/VttProperties.html#/c:@M@BitmovinPlayer@objc(cs)BMPVttProperties(py)vertical): `Returns the writing direction`.

Calling it `direction` or even `writingDirection` seemed more sensible to me than calling it `vertical`. Because it makes room for some strange use cases like:

```typescript
const cue: Cue = {
  vttPosition: {
    vertical: CueVertical.HORIZONTAL, // Why is the `vertical` prop receiving a value called `horizontal`?
    // ...
  },
};
```

Whilist `direction` would make a lot more sense:

```typescript
const cue: Cue = {
  vttPosition: {
    direction: CueDirection.HORIZONTAL, 
    // ...
  },
};
```

Anyway, those renames were just something that felt natural to me, but if you don't like them, I'll change them. No need to get worried.

I can simply rename everything back exactly to what they're called in the native code if that's what you want.

Reply:
Cues support has been dropped for now, see [commit](https://github.com/bitmovin/bitmovin-player-react-native/pull/14/commits/1599353362e30aa2471b60d7ddcff02d88bf8b2c).